main.tf

```diff
@@ -4,26 +4,30 @@ locals {
 
 # https://www.terraform.io/docs/providers/aws/r/vpn_gateway.html
 resource "aws_vpn_gateway" "default" {
-  count           = local.enabled ? 1 : 0
+  count           = var.vpn_gateway_id != null ? 0 : (local.enabled ? 1 : 0)
   vpc_id          = var.vpc_id
   amazon_side_asn = var.vpn_gateway_amazon_side_asn
   tags            = module.this.tags
 }
 
 # https://www.terraform.io/docs/providers/aws/r/customer_gateway.html
 resource "aws_customer_gateway" "default" {
-  count      = local.enabled ? 1 : 0
+  count      = local.enabled && var.customer_gateway_id == null ? 1 : 0
   bgp_asn    = var.customer_gateway_bgp_asn
   ip_address = var.customer_gateway_ip_address
   type       = "ipsec.1"
-  tags       = module.this.tags
+  tags = merge(
+    module.this.tags,
+    var.customer_gateway_name_override != null ? { Name = var.customer_gateway_name_override } : {},
+    { ip = var.customer_gateway_ip_address }
+  )
 }
 
 # https://www.terraform.io/docs/providers/aws/r/vpn_connection.html
 resource "aws_vpn_connection" "default" {
   count                    = local.enabled ? 1 : 0
-  vpn_gateway_id           = join("", aws_vpn_gateway.default.*.id)
-  customer_gateway_id      = join("", aws_customer_gateway.default.*.id)
+  vpn_gateway_id           = var.vpn_gateway_id != null ? var.vpn_gateway_id : join("", aws_vpn_gateway.default.*.id)
```

Comment:
I like to see this logic done elsewhere, particularly when it's reused like you do below. Let's pull this to a local like so: 

```hcl
locals {
	vpn_gateway_id = coalesce(var.vpn_gateway_id, join("", aws_vpn_gateway.default.*.id))
}
```

Then you can just reference the local and avoid the duplication of logic.